src/main/java/com/limechain/network/protocol/grandpa/GrandpaEngine.java

```diff
@@ -21,22 +21,31 @@
 
 @Log
 public class GrandpaEngine {
+    private static final GrandpaEngine INSTANCE = new GrandpaEngine();
```

Comment:
Why is this a singleton?

Reply:
This was a singleton due to synchronization logic, which was moved. No longer singleton.